### ISSUE TITLE
Error message for ncclTopoIdToIndex()

### ISSUE THREAD

**Issue by kwen2501** [NVIDIA/nccl#675]
Hi Team,

Can you please add an error message for the error returned here?

https://github.com/NVIDIA/nccl/blob/9bfc1c6e3587bd8f0dfadc42afd16dfad4ccd75c/src/graph/topo.h#L166

I saw a case hitting this path but did not see explicit error message:
```
hostname:2714030:2728529 [2] NCCL INFO graph/paths.cc:312 -> 3
hostname:2714030:2728529 [2] NCCL INFO transport/net.cc:140 -> 3
hostname:2714030:2728529 [2] NCCL INFO transport.cc:34 -> 3
hostname:2714030:2728529 [2] NCCL INFO transport.cc:84 -> 3
hostname:2714030:2728529 [2] NCCL INFO init.cc:778 -> 3
hostname:2714030:2728529 [2] NCCL INFO init.cc:904 -> 3
hostname:2714030:2728529 [2] NCCL INFO group.cc:72 -> 3 [Async thread]
```

Thank you!

**Comment by kwen2501** [comment]
Here is the full [log](https://gist.github.com/kwen2501/1d0345485ecd21b6dede90dabe7e935a) in case it helps to figure out the "internal error" reason better.

No need to debug the case though (it is sort-of a corner case as you may see from the env settings)

**Comment by sjeaugey** [comment]
The log should be something like:
```
  WARN("Could not find GPU in topology with rank %d", rank);
```
In the full log you provided I do not see the same stacktrace as above; it seems to be the trace of one rank which just could not connect to the bootstrap thread. Did I miss something or was the trace incomplete?

**Comment by sjeaugey** [comment]
Also, the link you 
Actually the link you provided was pointing to ncclTopoRankToIndex(). The function failing is indeed [ncclTopoIdToIndex](https://github.com/NVIDIA/nccl/blob/9bfc1c6e3587bd8f0dfadc42afd16dfad4ccd75c/src/graph/topo.h#L155) and the warning message should be:
```
  WARN("Could not find %s in topology with id %lx", topoNodeTypeStr[type], id);
```

**Comment by kwen2501** [comment]
Right, the later failures were just side effect that the other ranks could not connect to this failing rank.

Would you like the issue be closed via a PR or via the next release? In the later case, I can close the issue right now.

**Comment by sjeaugey** [comment]
I added the WARN in ncclTopoIdToIndex but had to remove it from ncclTopoRankToIndex because that function is sometimes used to check whether a rank is on the node or not and it's normal for it to not find the rank (ideally we'd return ncclSuccess and check rank!=-1 but that would need more work).

So it will arrive in a future release. If you need it now, we can also push a commit on github. Let us know.

**Comment by kwen2501** [comment]
Thanks Sylvain for the improvement. This is not urgent, next release is good. I am closing the issue now.

**Comment by sjeaugey** [comment]
> I added the WARN in ncclTopoIdToIndex
Actually I had to undo that change as well. It was causing WARNs on normal situations (e.g. when GPUs inside a node can't see each other and need to communicate through the network.

We need a bit more work here, to only have a WARN in abnormal situations.

**Comment by stephenmsachs** [comment]
### Issue Cleanup: Helping Us Focus on Current Challenges

  We're reviewing older issues to ensure we prioritize the most relevant and active ones. Since this issue hasn't seen updates in over 6 months, we'll be closing it for now.

  ,*This change helps us focus our efforts on addressing any current issues our users are facing.* If this issue still affects you, please don't hesitate to reopen it with a quick update (e.g., "Still relevant on [version=X]").
  Thanks for your understanding and for contributing to NCCL.
